googleapis_auth/test/oauth2_test.dart

```diff
@@ -325,19 +325,17 @@ void main() {
         final client = autoRefreshingClient(
             clientId,
             credentials,
-            mockClient(
-                expectAsync1((request) {
-                  if (serverInvocation++ == 0) {
-                    // This should be a refresh request.
-                    expect(request.headers['foo'], isNull);
-                    return successfulRefresh(request);
-                  } else {
-                    // This is the real request.
-                    expect(request.headers['foo'], equals('bar'));
-                    return Future.value(Response('', 200));
-                  }
-                }, count: 2),
-                expectClose: false));
+            mockClient(expectAsync1((request) {
+              if (serverInvocation++ == 0) {
+                // This should be a refresh request.
+                expect(request.headers['foo'], isNull);
```

Comment:
nit, can do `not(contains('foo'))` – I think

Reply:
I didn't write that, but since you pointed it out, I'll go ahead and fix. However, I'm not planning to go through existing code and fix any more nits tonight. :D

Reply:
ok, so `expect(request.headers, contains('foo'));` works -- but I couldn't find a `not` matcher, and writing a predicate to wrap the logic makes it appear even more obfuscated. It would be great to have a `not` matcher added to core_matchers, but I'm gonna pass on this nit for now. Tests are passing, I'm done for the night!

Reply:
no worries!